l1-contracts/src/core/interfaces/messagebridge/IOutbox.sol

```diff
@@ -1,65 +1,67 @@
  */
 interface IOutbox {
-  // to make it easier for portal to know when to consume the message.
-  event MessageAdded(bytes32 indexed entryKey);
+  event RootAdded(uint256 indexed l2BlockNumber, bytes32 indexed root, uint256 height);
+  event MessageConsumed(
+    uint256 indexed l2BlockNumber,
+    bytes32 indexed root,
+    bytes32 indexed messageHash,
+    uint256 leafIndex
+  );
 
-  event MessageConsumed(bytes32 indexed entryKey, address indexed recipient);
-
-  // docs:start:outbox_compute_entry_key
-  /**
-   * @notice Computes an entry key for the Outbox
-   * @param _message - The L2 to L1 message
-   * @return The key of the entry in the set
-   */
-  function computeEntryKey(DataStructures.L2ToL1Msg memory _message) external returns (bytes32);
-  // docs:end:outbox_compute_entry_key
-
-  // docs:start:outbox_send_l1_msg
+  // docs:start:outbox_insert
   /**
-   * @notice Inserts an array of entries into the Outbox
+   * @notice Inserts the root of a merkle tree containing all of the L2 to L1 messages in
+   * a block specified by _l2BlockNumber.
    * @dev Only callable by the rollup contract
-   * @param _entryKeys - Array of entry keys (hash of the message) - computed by the L2 counterpart and sent to L1 via rollup block
+   * @dev Emits `RootAdded` upon inserting the root successfully
+   * @param _l2BlockNumber - The L2 Block Number in which the L2 to L1 messages reside
+   * @param _root - The merkle root of the tree where all the L2 to L1 messages are leaves
+   * @param _height - The height of the merkle tree that the root corresponds to
    */
-  function sendL1Messages(bytes32[] memory _entryKeys) external;
-  // docs:end:outbox_send_l1_msg
+  function insert(uint256 _l2BlockNumber, bytes32 _root, uint256 _height) external;
+  // docs:end:outbox_insert
 
   // docs:start:outbox_consume
   /**
    * @notice Consumes an entry from the Outbox
-   * @dev Only meaningfully callable by portals, otherwise should never hit an entry
-   * @dev Emits the `MessageConsumed` event when consuming messages
+   * @dev Only useable by portals / recipients of messages
+   * @dev Emits `MessageConsumed` when consuming messages
+   * @param _l2BlockNumber - The block number specifying the block that contains the message we want to consume
+   * @param _leafIndex - The index inside the merkle tree where the message is located
    * @param _message - The L2 to L1 message
-   * @return entryKey - The key of the entry removed
+   * @param _path - The sibling path used to prove inclusion of the message, the _path length directly depends
+   * on the total amount of L2 to L1 messages in the block. i.e. the length of _path is equal to the depth of the
+   * L1 to L2 message tree.
    */
-  function consume(DataStructures.L2ToL1Msg memory _message) external returns (bytes32 entryKey);
+  function consume(
+    uint256 _l2BlockNumber,
+    uint256 _leafIndex,
+    DataStructures.L2ToL1Msg calldata _message,
```

Comment:
As mentioned earlier, we could consider if it is nicer with the "message, proof" split 🤷